app/routes.py

```diff
@@ -1 +1,302 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, make_response, abort
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import date
+
+
+
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+                            #### HELPER FUNCTIONS ####
+
+def return_task_or_abort(task_id):
+    try:
+        verified_id = int(task_id)
+    except ValueError:
+        abort(make_response({"details":"Invalid ID: id must be an integer"}, 400))
+
+    task = Task.query.get(verified_id)
+
+    if not task:
+        abort(make_response({"details":"Invalid ID: id does not exist"}, 404))
+
+    return task
+
+
+
+def return_goal_or_abort(goal_id):
+    try:
+        verified_id = int(goal_id)
+    except ValueError:
+        abort(make_response({"details":"Invalid ID: id must be an integer"}, 400))
+
+    goal = Goal.query.get(verified_id)
+
+    if not goal:
+        abort(make_response({"details":"Invalid ID: id does not exist"}, 404))
+
+    return goal
+
+
+def boolean_completeness(completed_at):
+    if not completed_at:
+        return False
+    return True
+
+
+def format_return_task(target_task):
+    return {"id": target_task.task_id,
+            "goal_id": target_task.goal_id,
+            "title": target_task.title,
+            "description": target_task.description,
+            "is_complete": boolean_completeness(target_task.completed_at)
+            }
+    
+def format_return_goal(target_goal):
+    return {"id": target_goal.goal_id,
+            "title": target_goal.title
+            }
+
+
+
+
+
+                                #### TASK ROUTES ####
+
+@tasks_bp.route("", methods = ["POST"])
+def add_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(
+            title=request_body["title"],
+            description=request_body["description"]
+            # completed_at=request_body["completed_at"],
+        )
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {"task":format_return_task(new_task)}, 201
+    except KeyError:
+        return {"details":"Invalid data"}, 400
```

Comment:
It's good that you're doing input validation here, but it would be better to indicate to the user what/which data is invalid.